### PR TITLE
ci: upload test results as an artifact

### PR DESCRIPTION
**Pull request by sadlerap**
Since we run tests in parallel, behavex doesn't output a lot to the console.  To retain some sense of ability to debug testing failures in CI, we should upload the output of behavex as an artifact to be inspected.

### CODE REVIEW

**Comment by scothis** [comment]
As discussion on the working group call. We need to make sure the output artifact is capture for each test run (it's currently overwriting). Another option is to dump the logs to the console in the diagnostics step.

**Comment by scothis** [comment]
Not required for this PR, but using something like https://github.com/marketplace/actions/junit-report-action could make the test reports more accessible without downloading the zip

**Comment by scothis** [comment]
@sadlerap can we merge this?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -359,6 +366,12 @@ jobs:
         set -o nounset
         set +o pipefail
 
+        if [ -e conformance-tests/test-output/results/failing_scenarios.txt]
+          echo "##[group]failing tests"
+          cat conformance-tests/test-output/results/failing_scenarios.txt
+          echo "##[endgroup]"
+        fi
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply .github/workflows/ci.yaml]: Fixed in 92583bd

**Comment by sadlerap** [comment]
@scothis Should be good to merge at this point IMO, unless you have any particular reason to hold up on this.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -359,6 +366,12 @@ jobs:
         set -o nounset
         set +o pipefail
 
+        if [ -e conformance-tests/test-output/results/failing_scenarios.txt] ; then
```
Comment (scothis) [inline .github/workflows/ci.yaml:369]:
```suggestion
        if [ -e conformance-tests/test-output/results/failing_scenarios.txt ] ; then
```

**Pull request merged**